### PR TITLE
Solaris: fix assert for betterC build.

### PR DESCRIPTION
**Pull request by nykytenko**

### CODE REVIEW

File: gen/llvmhelpers.cpp [hunk 1]
```diff
@@ -311,6 +311,14 @@ void DtoCAssert(Module *M, Loc &loc, LLValue *msg) {
     args.push_back(file);
     args.push_back(line);
     args.push_back(msg);
+  } else if (global.params.targetTriple->isOSSolaris()) {
+    const auto irFunc = gIR->func();
+    const auto funcName =
+        irFunc && irFunc->decl ? irFunc->decl->toPrettyChars() : "";
```
Comment (JohanEngelen) [inline gen/llvmhelpers.cpp]: can you add parens here, to avoid relying on operator precedence rules. (i know it is a copy from a few lines above)

**Comment by JohanEngelen** [comment]
LGTM (with the nit). A lit test would be nice for betterC assert function name (do we have one already?).

**Comment by kinke** [comment]
A [DMD test](https://github.com/ldc-developers/dmd-testsuite/blob/ldc/runnable/betterc.d) makes sure a betterC program with assertions can be linked.

File: gen/llvmhelpers.cpp (hunk 1 above)
Comment (nykytenko) [reply gen/llvmhelpers.cpp]: Done.

**Pull request merged**